var/spack/repos/builtin/packages/py-libensemble/package.py

```diff
@@ -23,13 +24,21 @@ class PyLibensemble(PythonPackage):
     version('0.2.0', 'ee96047594a3f5a1533f24d3b1f365f9')
     version('0.1.0', '0c3d45dd139429de1a5273e5bd8e46ec')
 
-    depends_on('python@3.4:', when='@0.5.0:')
-    depends_on('python@2.7:2.8,3.3:', when='@:0.4.1')
+    variant('mpi',  default=False, description='Install with MPI')
+    variant('scipy',  default=False, description='Install with scipy')
+    variant('petsc4py',  default=False, description='Install with petsc4py')
+    variant('nlopt',  default=False, description='Install with nlopt')
+
+    # depends_on('python@2.7:2.8,3.3:', when='@:0.4.1')
+    # depends_on('python@3.5:', when='@0.5.0:')
+    depends_on('python@3.5:')
     depends_on('py-setuptools', type='build')
-    depends_on('mpi')
-    depends_on('py-mpi4py@2.0:', type=('build', 'run'))
     depends_on('py-numpy', type=('build', 'run'))
-    depends_on('py-scipy', type=('build', 'run'))
-    depends_on('py-petsc4py', type=('build', 'run'))
-    depends_on('py-petsc4py@develop', type=('build', 'run'), when='@develop')
-    depends_on('nlopt', type=('build', 'run'))
+    depends_on('mpi', when='@:0.4.1')
```

Comment:
@shuds13 I wanted to make sure: should this always depend on `mpi` when the version is <= `0.4.1`?

If building with `mpi` is not optional before that version, you could also add a `conflicts` statement:

```
conflicts('~mpi`, when='@:0.4.1')
```

Reply:
Yes, it requires mpi up to version 0.4.1. I guess by using depends_on it automatically brings it in, but with conflicts you have to add the +mpi? I'm happy to change, if that is what you recommend.